internal/reconciler/shoot/reconciler.go

```diff
@@ -23,10 +40,125 @@ type Reconciler struct {
+		return ctrl.Result{RequeueAfter: r.ResyncPeriod}, nil
+	}
+
+	// TODO(theoddora): Add proper validation that a single issuer is not registered more than once
+	// This should check if another OIDC resource with the same issuerURL already exists
+
+	userNameClaim := "sub"
+	groupsClaim := "groups"
+	prefix := buildPrefix(shoot)
+	userNamePrefix := prefix
+	groupsPrefix := prefix
+
+	oidc := emptyOIDC(shoot)
+	if _, err := controllerutils.GetAndCreateOrMergePatch(ctx, r.Client, oidc, func() error {
+		oidc.Annotations = utils.MergeStringMaps(oidc.Annotations, map[string]string{})
+		oidc.Labels = utils.MergeStringMaps(oidc.Labels, map[string]string{
+			LabelKeyManagedBy: LabelValueGardenShootTrustConfigurator,
+		})
+		oidc.Spec = authenticationv1alpha1.OIDCAuthenticationSpec{
+			IssuerURL: issuerURL,
+			// TODO(theoddora): Consider configuring ClientID based on landscape
+			ClientID:       "garden",
+			UsernameClaim:  &userNameClaim,
+			UsernamePrefix: &userNamePrefix,
+			GroupsClaim:    &groupsClaim,
+			GroupsPrefix:   &groupsPrefix,
+		}
+		return nil
+	}); err != nil {
+		log.Error(err, "Failed to create or update OIDC resource", "oidc", client.ObjectKeyFromObject(oidc))
+		return ctrl.Result{RequeueAfter: r.ResyncPeriod}, err
+	}
+
+	log.Info("Successfully created or updated OIDC resource for shoot", "oidc", client.ObjectKeyFromObject(oidc))
+	return ctrl.Result{RequeueAfter: r.ResyncPeriod}, nil
+}
+
+func (r *Reconciler) deleteOIDC(ctx context.Context, log logr.Logger, shoot *gardencorev1beta1.Shoot) (ctrl.Result, error) {
+	oidc := emptyOIDC(shoot)
+	err := r.Client.Get(ctx, types.NamespacedName{Name: getOIDCResourceName(shoot), Namespace: v1beta1constants.GardenNamespace}, oidc)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Info("OIDC resource not found, nothing to do", "oidc", client.ObjectKeyFromObject(oidc))
+			return reconcile.Result{}, nil
+		}
+		log.Error(err, "Failed to get OIDC resource", "oidc", client.ObjectKeyFromObject(oidc))
+		return reconcile.Result{}, fmt.Errorf("failed to get OIDC: %w", err)
+	}
+
+	if err := r.Client.Delete(ctx, oidc); err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Info("OIDC resource not found, nothing to do", "oidc", client.ObjectKeyFromObject(oidc))
+			return reconcile.Result{}, nil
+		}
+		log.Error(err, "Failed to delete OIDC resource", "oidc", client.ObjectKeyFromObject(oidc))
+		return ctrl.Result{RequeueAfter: r.ResyncPeriod}, fmt.Errorf("failed to delete OIDC: %w", err)
+	}
+	log.Info("Successfully deleted OIDC resource", "oidc", client.ObjectKeyFromObject(oidc))
+	return reconcile.Result{}, nil
+}
+
+func emptyOIDC(shoot *gardencorev1beta1.Shoot) *authenticationv1alpha1.OpenIDConnect {
+	return &authenticationv1alpha1.OpenIDConnect{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      getOIDCResourceName(shoot),
+			Namespace: v1beta1constants.GardenNamespace,
```

Comment:
No need to set namespace on these ones as they are cluster scoped and are not actually namespaced.
```suggestion
```